lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_betabuild_prechecks.rb

```diff
@@ -8,8 +8,9 @@ def self.run(params)
         require_relative '../../helper/android/android_version_helper'
         require_relative '../../helper/android/android_git_helper'
 
-        # Checkout develop and update
-        Fastlane::Helper::GitHelper.checkout_and_pull('develop')
+        # Checkout default branch and update
+        default_branch = params[:default_branch] || 'develop'
```

Comment:
@oguzkocer you might have to explicitly define a new `FastlaneCore::ConfigItem` in the `available_options` below (line 72) in order to use this.

Fastlane has it's own mechanism to manage the parameters it sends to the actions, but Ruby's deceiving simplicity doesn't make it clear.

I verified this by adding a similar line to an action that has a test and got errors due to it:

```diff
--- a/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb
+++ b/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb
@@ -2,6 +2,7 @@ module Fastlane
   module Actions
     class IosGenerateStringsFileFromCodeAction < Action
       def self.run(params)
+        foo = params[:foo] || 'foo'
         files = files_matching(paths: params[:paths], exclude: params[:exclude])
         flags = [('-q' if params[:quiet]), ('-SwiftUI' if params[:swiftui])].compact
         flags += Array(params[:routines]).flat_map { |routine| ['-s', routine] }
```

Errors when running the tests:

```
     Failure/Error: foo = params[:foo] || 'foo'

     NoMethodError:
       undefined method `fetch_env_value' for nil:NilClass
```

After adding the value, the tests passed as usual. 

```diff
--- a/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb
+++ b/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb
@@ -2,6 +2,7 @@ module Fastlane
   module Actions
     class IosGenerateStringsFileFromCodeAction < Action
       def self.run(params)
+        foo = params[:foo] || 'foo'
         files = files_matching(paths: params[:paths], exclude: params[:exclude])
         flags = [('-q' if params[:quiet]), ('-SwiftUI' if params[:swiftui])].compact
         flags += Array(params[:routines]).flat_map { |routine| ['-s', routine] }
@@ -90,6 +91,10 @@ module Fastlane
                                        description: 'If true, will fail with user_error! if `genstrings` printed any error while parsing',
                                        is_string: false, # Boolean
                                        default_value: true),
+          FastlaneCore::ConfigItem.new(key: :foo,
+                                       env_name: 'FL_IOS_GENERATE_STRINGS_FILE_FROM_CODE_FOO',
+                                       description: 'foo',
+                                       default_value: 'foo'),
         ]
       end
 ```

This approach has the advantage of moving the default value, or rather fallback if the user doesn't set it, to the `ConfigItem` object, so the caller code becomes a bit clearer as one merely needs to run `params[:foo]`.

Reply:
@mokagio Thanks a lot for pointing this out to me. I implemented the change in 8c510ea458ca3ccb176d4d85c3df969733de0e05, I hope that's correct. Let me know what you think! cc @AliSoftware (due to the other discussions we have had in the comments and on Slack)